services/src/main/java/io/grpc/services/health/HealthStatusManager.java

```diff
@@ -0,0 +1,96 @@
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ *    * Redistributions of source code must retain the above copyright
+ * notice, this list of conditions and the following disclaimer.
+ *    * Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following disclaimer
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *
+ *    * Neither the name of Google Inc. nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.grpc.services.health;
+
+import static io.grpc.services.health.HealthOuterClass.HealthCheckResponse.ServingStatus;
+
+import javax.annotation.Nullable;
+
+/**
+ * A {@code HealthStatusManager} object manages a health check service. A health check service is
+ * created in the constructor of {@code HealthStatusManager}, and it can be retrieved by the
+ * {@link #getHealthService()} method.
+ * The health status manager can update the health status of gRPC services.
+ */
+@io.grpc.ExperimentalApi
+public final class HealthStatusManager {
+
+  private final HealthServiceImpl healthService;
+
+  /**
+   * Creates a new health service instance.
+   */
+  public HealthStatusManager() {
+    healthService = new HealthServiceImpl();
+  }
+
+  /**
+   * Gets the health check service created in the constructor.
+   */
+  public HealthGrpc.AbstractHealth getHealthService() {
+    return healthService;
+  }
+
+  /**
+   * Updates the status of service.
+   * @param status is one of the values {@link ServingStatus#SERVING},
+   *     {@link ServingStatus#NOT_SERVING} and {@link ServingStatus#UNKNOWN}.
+   */
+  public void setStatus(String service, ServingStatus status) {
+    healthService.setStatus(service, status);
+  }
+
+  /**
+   * Gets the health status of a service.
+   * @return {@code null} if there is no record of the status of the service or the service does
+   *     not exist.
+   */
+  @Nullable
+  public ServingStatus getStatus(String service) {
```

Comment:
I think we need, because this could be more efficient for the status updater (who calls setStatus method, and is not necessary to invoke rpc request) to get the current status before setting new status. 

If through health check rpc request, you have to create a client first, wrap up request message, implement response observer, and try and catch.